platforms/windows/SQLitePlugin/SQLitePlugin.cpp

```diff
@@ -0,0 +1,511 @@
+#include "pch.h"
+
+#include "SQLitePlugin.h"
+
+using namespace winrt;
+using namespace winrt::Microsoft::ReactNative;
+using namespace winrt::Windows::Foundation;
+using namespace winrt::Windows::Storage;
+using namespace winrt::Windows::Storage::Streams;
+using namespace winrt::Windows::Security::Cryptography;
+
+namespace SQLitePlugin
+{
+    void SQLitePlugin::Attach(
+        JSValue options,
+        std::function<void(std::string)> onSuccess,
+        std::function<void(std::string)> onFailure) noexcept
+    {
+        SerialReactDispatcher.Post(
+            [onFailure{ std::move(onFailure) }]()
+        {
+            onFailure("attach isn't supported on this platform");
+        });
+    }
+
+    void SQLitePlugin::Close(
+        CloseOptions options,
+        std::function<void(std::string)> onSuccess,
+        std::function<void(std::string)> onFailure) noexcept
+    {
+        SerialReactDispatcher.Post(
+            [
+                options{ std::move(options) },
+                onSuccess{ std::move(onSuccess) },
+                onFailure{ std::move(onFailure) },
+                weak_this = std::weak_ptr(shared_from_this())
+            ]()
+        {
+
+            if (auto strongThis = weak_this.lock()) {
+
+                if (options.Path == nullptr || options.Path.empty())
+                {
+                    onFailure("You must specify database path");
+                    return;
+                }
+
+                hstring absoluteDbPath{ ResolveDbFilePath(to_hstring(options.Path)) };
+
+                if (strongThis->OpenDBs.find(absoluteDbPath) == strongThis->OpenDBs.end())
+                {
+                    onFailure("Specified db was not open");
+                    return;
+                }
+
+                sqlite3* dbHandle = strongThis->OpenDBs[absoluteDbPath];
+
+                if (sqlite3_close(dbHandle) != SQLITE_OK)
+                {
+                    // C# implementation returns success if the DB failed to close
```

Comment:
How are you 'returning success' on failure here? You're just logging a failure msg, right?

Reply:
Also, if sth is not right in the legacy implementation, this is a good time to fix it, imo.

Reply:
I think the benchmark to match it other platforms in this repo rather, than the legacy c# impl.